### DISCUSSION TITLE
UPDATE (Windows Terminal appears to be the issue): How to get item documentation to show and why isn't C-space triggering anything?

### DISCUSSION THREAD

**Discussion by Or1g3n** [saghen/blink.cmp#213] [General]
Hello, 

I am having a hard time understanding how to get item docs to show when selecting from completion list. In addition, the keymaps that are set to <C-Space> do not appear to do anything.

I know documentation provided by the LSP is working because after a function is selected, pressing K in normal mode, shows the function docs as expected. However, docs are not auto-showing during item selection even though I believe I have set this property to true (see config below).

In regards to the <C-Space> issue, when I start typing auto-completion happens by itself which is great but pressing <C-Space> does not appear to manually trigger it if I do so at the end of an incomplete method (os.da. for example).

Any help on these items would be greatly appreciated. I have already spent a few hours trying to troubleshoot and reviewed random repos from others who use blink.cmp in their nvim configs but not seeing anything different than what I have below.

In case this is relevant:
OS: Windows 10
Terminal: Windows Terminal
Shell: Nushell

Below is my configuration for blink.cmp (below this is an image from a lua file showing that item docs are not present):
``` lua
return {
    'saghen/blink.cmp',
    lazy = false, -- lazy loading handled internally
    -- optional: provides snippets for the snippet source
    dependencies = 'rafamadriz/friendly-snippets',
    -- use a release tag to download pre-built binaries
    version = 'v0.*',
    -- OR build from source, requires nightly: https://rust-lang.github.io/rustup/concepts/channels.html#working-with-nightly-rust
    -- build = 'cargo build --release',
    -- If you use nix, you can build from source using latest nightly rust with:
    -- build = 'nix run .#build-plugin',

    ---@module 'blink.cmp'
    ---@type blink.cmp.Config
    opts = {
	highlight = {
	    -- sets the fallback highlight groups to nvim-cmp's highlight groups
	    -- useful for when your theme doesn't support blink.cmp
	    -- will be removed in a future release, assuming themes add support
	    use_nvim_cmp_as_default = true,
	},
	-- set to 'mono' for 'Nerd Font Mono' or 'normal' for 'Nerd Font'
	-- adjusts spacing to ensure icons are aligned
	nerd_font_variant = 'mono',

	-- experimental auto-brackets support
	-- accept = { auto_brackets = { enabled = true } },

	-- experimental signature help support
	-- trigger = { signature_help = { enabled = true } },

	-- for keymap, all values may be string | string[]
	-- use an empty table to disable a keymap
	keymap = {
	    show = '<C-Space>',
	    hide = '<C-e>',
	    accept = '<Tab>',
	    select_and_accept = {},
	    select_prev = { '<Up>', '<C-p>', '<C-k>' },
	    select_next = { '<Down>', '<C-n>', '<C-j>' },

	    show_documentation = '<C-Space>',
	    hide_documentation = '<C-Space>',
	    scroll_documentation_up = '<C-b>',
	    scroll_documentation_down = '<C-f>',

	    snippet_forward = '<Tab>',
	    snippet_backward = '<S-Tab>',
	},
	documentation = {
	    min_width = 10,
	    max_width = 60,
	    max_height = 20,
	    border = 'padded',
	    winhighlight = 'Normal:BlinkCmpDoc,FloatBorder:BlinkCmpDocBorder,CursorLine:BlinkCmpDocCursorLine,Search:None',
	    -- which directions to show the documentation window,
	    -- for each of the possible autocomplete window directions,
	    -- falling back to the next direction when there's not enough space
	    direction_priority = {
		autocomplete_north = { 'e', 'w', 'n', 's' },
		autocomplete_south = { 'e', 'w', 's', 'n' },
	    },
	    -- Controls whether the documentation window will automatically show when selecting a completion item
	    auto_show = true,
	    auto_show_delay_ms = 500,
	    update_delay_ms = 50,
	},
    }
}
```

![image](https://github.com/user-attachments/assets/9cbf59f7-3723-4c08-be5d-e98047771d3d)

**Comment by yngwi** [comment]
I also notice the trigger problem with ctrl-space and think it is related to the windows terminal. When using it in Neovide, the mapping works as expected so I think it might actually be windows terminal that swallows the event before it reaches nvim.

**Reply by Or1g3n** [reply]
@yngwi Daniel, you are an angel with wings lol. I just tried in Neovide and both issues are now resolved ( c-space now works and documentation on item selection is also working ). Hopefully there is a way to resolve the conflict with windows terminal!

Thanks a ton for the reply!
